src/main/kotlin/com/ekino/oss/recaptcha/config/ReCaptchaProperties.kt

```diff
@@ -0,0 +1,30 @@
+package com.ekino.oss.recaptcha.config
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+import org.springframework.http.HttpMethod
+import org.springframework.http.HttpMethod.POST
+import java.time.Duration
+
+const val DEFAULT_RESPONSE_PARAM_NAME = "g-recaptcha-response"
+private const val GOOGLE_RECAPTCHA_URL = "https://www.google.com/recaptcha/api/"
+private const val TIMEOUT_DEFAULT_SECONDS = 5L
+private val defaultTimeout = Duration.ofSeconds(TIMEOUT_DEFAULT_SECONDS)
+
+@ConfigurationProperties(prefix = "security.recaptcha")
+@ConstructorBinding
+data class ReCaptchaProperties(
+  val client: ClientProperties = ClientProperties(),
+  val responseName: String = DEFAULT_RESPONSE_PARAM_NAME,
+  val secret: String,
```

Comment:
If the user does not give a secret, is Spring able to say by itself that we require a non-null value ? In Java we would need to use `@Validated` on the class and `@NotNull` on the field in order to have a clear error on application startup, but I don't know if it is necessary and how it would be done in kotlin

Reply:
Yes the application fail to start if we don't provide the secret : `Caused by: java.lang.IllegalArgumentException: Parameter specified as non-null is null: method com.ekino.oss.recaptcha.config.ReCaptchaProperties.<init>, parameter secret`

Reply:
It would be nice if Spring could detect that the parameter is non-null, and return a meaningful exception instead of trying at runtime to pass a null value to a non-null argument. I'll try asking for a feature request and resolve this in the mean time 😛